tests/src/test/java/alluxio/server/ft/BackwardsCompatibilityTest.java

```diff
@@ -0,0 +1,525 @@
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.server.ft;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import alluxio.AlluxioTestDirectory;
+import alluxio.AlluxioURI;
+import alluxio.Constants;
+import alluxio.ProjectConstants;
+import alluxio.PropertyKey;
+import alluxio.client.WriteType;
+import alluxio.client.file.FileOutStream;
+import alluxio.client.file.FileSystem;
+import alluxio.client.file.FileSystemMasterClient;
+import alluxio.client.file.URIStatus;
+import alluxio.client.file.options.CreateFileOptions;
+import alluxio.client.file.options.SetAttributeOptions;
+import alluxio.master.MasterClientConfig;
+import alluxio.master.file.meta.PersistenceState;
+import alluxio.multi.process.MultiProcessCluster;
+import alluxio.multi.process.MultiProcessCluster.Builder;
+import alluxio.multi.process.PortCoordination;
+import alluxio.security.authorization.AclEntry;
+import alluxio.util.CommonUtils;
+import alluxio.util.WaitForOptions;
+import alluxio.util.io.PathUtils;
+import alluxio.wire.SetAclAction;
+
+import com.google.common.collect.ImmutableList;
+import org.apache.commons.io.FileUtils;
+import org.hamcrest.CoreMatchers;
+import org.junit.After;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * Tests that the current version can read journals from previous versions.
+ *
+ * This class has a main() method which can be used to populate src/test/resources/old_journals
+ * with a journal and journal backup built from the current version. Whenever we release, we
+ * generate new journal artifacts for the released version. The readOldJournals test will iterate
+ * over all journal artifacts, replay them using the current version, and verify the results.
+ *
+ * To cover many types of journal entries, the test runs a series of TestOps, each of which makes
+ * some independent modification to master state, and has a method to verify that the change was
+ * correctly applied. To add a new TestOp, implement the TestOp interface or extend FsTestOp, and
+ * add your TestOp to the OPS list. Also, either re-generate all journal artifacts from previous
+ * versions (not easy at the moment, could add tooling for this later), or implements
+ * supportsVersion to only match the latest version and future versions.
+ */
+@Ignore("Test is on the slow side, and doesn't need to be run with every build")
```

Comment:
Should we ignore this at the maven level and have a profile which runs it?

Reply:
good idea. I'll introduce a `skipSlowTests` profile which is active by default, and de-activate it for the master build.